src/nmodl/main.cpp

```diff
@@ -587,7 +617,16 @@ int main(int argc, const char* argv[]) {
             }
 
 #ifdef NMODL_LLVM_BACKEND
-            if (llvm_ir) {
+
+            if (run_benchmark) {
+                logger->info("Running LLVM benchmark");
+                benchmark::LLVMBuildInfo info{llvm_vec_width, llvm_opt_passes, llvm_float_type};
+                benchmark::LLVMBenchmark bench(
+                    modfile, output_dir, info, repeat, instance_size, backend);
```

Comment:
Maybe it would make sense to set the `llvm_vec_width` based on `backend` if vector width value is not defined by the user?

Reply:
The default vector width is 1 in our visitor, but I see what you mean! That is a good point, however, I think that for benchmarking we want a greater control of parameters (e.g. looking at different vector widths for a specific backend: LLVM can split long vectors into shorter ones if they are not supported by the target, and that can be interesting to analyse). 

@iomaganaris I guess we can have a (warning?) message saying that there is a vector length and target ISA mismatch?

Reply:
I see. I think that for having greater flexibility you may want to leave it as it is now and maybe add a `todo` to fix it later :+1:

Reply:
LGTM